gsy_framework/utils.py

```diff
@@ -585,3 +590,36 @@ def use_default_if_null(input_value: Union[int, str, float], default_value: Unio
                         ) -> Union[int, str, float]:
     """Return input_value if not null, else return provided default_value"""
     return input_value if input_value not in NULL_VALUES else default_value
+
+
+def _generate_time_slot_list(slot_length: timedelta, sim_duration: timedelta,
+                             start_date: datetime) -> List[datetime]:
+    return [start_date.add(minutes=slot_length.seconds / 60 * time_diff) for time_diff in
+            range(int(sim_duration.total_seconds() / slot_length.total_seconds()))]
+
+
+def resample_hourly_energy_profile(
+        input_profile: Dict[datetime, float], slot_length: timedelta, sim_duration: timedelta,
+        start_date: datetime) -> Dict[datetime, float]:
+    """Resample hourly energy profile in order to fit to the set slot_length."""
+    slot_length_minutes = slot_length.total_seconds() / 60
+    if slot_length_minutes < 60:
+        assert 60 % slot_length_minutes == 0, "slot_length is not division of 1 hour"
+        scaling_factor = 60 / slot_length_minutes
+        out_dict = {}
+        for time_slot in _generate_time_slot_list(slot_length, sim_duration, start_date):
+            hour_time_slot = datetime(time_slot.year, time_slot.month, time_slot.day,
+                                      time_slot.hour)
+            out_dict[time_slot] = input_profile[hour_time_slot] / scaling_factor
```

Comment:
The approach of generating all expected time slots and checking out whether they exist in the profile is fine by me. However, I am not sure that KeyError is the right way to react in case of missing datapoints. I would recommend to either wrap this line around KeyError, or to check whether the hour_time_slot is part of the input_profile. And if it is not, then raise a custom exception (maybe GSyProfileError like on the other comment) that the caller code can then catch.

Reply:
OK, this will prevent us from rebase returning profiles with holes.